modules/slider/Slider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import { withStyles } from "@diana-ui/base";
+import { ThemeStyleSheetFactory, WithStylesProps } from "@diana-ui/types";
+import { LabelRegular } from "@diana-ui/typography";
+
+export interface ISliderProps {
+  min: number;
+  max: number;
+  value: number;
+  step: number;
+  onChange: any;
+  disabled?: boolean;
+  className?: string;
+}
+
+export interface ISliderState {
+  leftSpacing: number;
+}
+
+const styleSheet: ThemeStyleSheetFactory = theme => ({
+  wrapper: {},
+  valueWrapper: {},
+  input: {
+    "&::-webkit-slider-thumb": {},
+    "@selectors": {
+      "&.disabled::-webkit-slider-thumb": {},
+      "&.disabled::-webkit-slider-runnable-track": {}
+    },
+    "&::-moz-range-thumb": {},
+    "&::-ms-thumb": {},
+    "&:focus": {},
+    "&::-ms-track": {},
+    "&::-webkit-slider-runnable-track": {}
+  },
+  value: {}
+});
+
+const calculateLeftSpace = (
+  totalWidth: number,
+  min: number,
+  max: number,
+  value: number
+) => (value * totalWidth) / (max - min) + 13 - (`${max}`.length * 20) / 2;
+
+const Slider: React.FC<ISliderProps & WithStylesProps> = ({
+  cx,
+  styles,
+  min,
+  max,
+  value,
```

Comment:
My opinion here, I prefer to let this components have their own controlled state (or as we call it, to be uncontrolled). 
This means, if you dont pass value, have something like:

const [_value, setValue] = useState(initialValue);

So it would always have its own value. This way we can move the slider and fire onChanges (if needed) without the need of the parent to pass the Slider the value itself. 

Of course we must keep the option of the value to be passed to the component as it is.
Usually as:

useEffect(() => {
if(value !== undefined)
setValue(value)
})

this way if it has a value, it becomes controlled by their parent.